analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Branching.LearnConstraint.cs

```diff
@@ -632,6 +632,38 @@ public void Branching_LearnsObjectConstraint_IsNull()
                 .And.ContainSingle(x => x.HasConstraint(ObjectConstraint.NotNull));
         }
 
+        [DataTestMethod]
+        [DataRow("arg.HasValue")]
+        [DataRow("arg.HasValue is true")]
```

Comment:
Doens't work, our general pattern matching doesn't support this

Reply:
That is unfortunate. It might also explain the problem reported here (even though this is about `is not null`):
https://community.sonarsource.com/t/c-s2259-is-null-on-at-least-one-execution-path-false-negatives/84152
We should keep this on our radar.